<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Dean Adams, Michael Collyer, Antigoni Kaliontzopoulou, &amp; Mark A. Conaway" />


<title>Estimate Missing Landmarks</title>

<script src="site_libs/header-attrs-2.19/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="style/tutorial_theme.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"><img id="logo"style="height: 25px;" src="geomorph.png" /></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    GM Preliminaries
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="1.1-GeomorphDataTypes.html">Geomorph Data Types</a>
    </li>
    <li>
      <a href="1.2-ReadingData.html">Reading Data</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Data Manipulation</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="1.3.1-GeomorphDataFrame.html">Geomorph Data Frame</a>
        </li>
        <li>
          <a href="1.3.2-CombineLmkConfig.html">Combine Landmark Configurations (combine.subsets)</a>
        </li>
        <li>
          <a href="1.3.3-Interlmkdist.html">Interlandmark Distances (interlmkdist)</a>
        </li>
        <li>
          <a href="1.3.4-Flip-RotateLmks.html">Rotate or Flip Landmark Configurations (rotate.coords)</a>
        </li>
        <li>
          <a href="1.3.5-AllAboutArrays.html">Create 3D Array (arrayspecs)</a>
        </li>
        <li>
          <a href="1.3.6-Two.d.array.html">Convert Array to Matrix (two.d.array)</a>
        </li>
        <li>
          <a href="1.3.7-EstMissing.html">Estimating Missing Landmarks</a>
        </li>
        <li>
          <a href="1.3.8-IdentifyOutliers.html">Outlier Identification</a>
        </li>
        <li>
          <a href="1.3.9-StandardizeArticulations.html">Standardizing Articulations</a>
        </li>
        <li>
          <a href="1.3.10-SymmetricShapeGeneration.html">Symmetric Shape Generation</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">R Data Basics</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="1.4.1-SpecimenNames.html">Adding &amp; Changing Specimen Names</a>
        </li>
        <li>
          <a href="1.4.2-ExtractingClassifiers.html">Extracting Classifiers</a>
        </li>
        <li class="dropdown-header">Reflecting Landmarks</li>
        <li>
          <a href="1.4.3-SwappingLandmarks.html">Swapping Landmarks</a>
        </li>
        <li>
          <a href="1.4.4-DeletingSpecLmk.html">Delete Specs. and Landmarks</a>
        </li>
        <li>
          <a href="1.4.5-MatchSpectoPhy.html">Matching Data to a Phylogeny</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Procrustes Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="2.1-GPAIntro.html">GPA Introduction</a>
    </li>
    <li>
      <a href="2.2-GPAfixedlmks.html">GPA with fixed landmarks</a>
    </li>
    <li>
      <a href="2.3-GPAcurves.html">GPA with curve semilandmarks</a>
    </li>
    <li>
      <a href="2.4-GPAsurface.html">GPA with surface semilandmarks</a>
    </li>
    <li class="dropdown-header">gpagen Advanced Options</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analytics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Linear Models</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="3.1.1-ANOVARegression.html">ANOVA/Regression (ProcD.lm)</a>
        </li>
        <li>
          <a href="3.1.2-Phylo-ANOVA-Regression.html">Phylogenetic ANOVA/Regression (ProcD.pgls)</a>
        </li>
        <li>
          <a href="3.1.3-Trajectoryanalysis.html">Phenotypic Trajectory Analysis</a>
        </li>
        <li>
          <a href="3.1.4-Pairwisecomp.html">Pairwise Comparisons (pairwise)</a>
        </li>
        <li>
          <a href="3.1.5-TwoBPLS.html">Two-block Partial Least Squares (two.b.pls)</a>
        </li>
        <li>
          <a href="3.1.6-Bilatsymmanalysis.html">Analysis of Bilateral Symmetry (bilat.symmetry)</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Shape Patterns</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="3.2.1-twobpls.html">Two-block Partial Least Squares (two.b.pls)</a>
        </li>
        <li>
          <a href="3.2.2-integrationtest.html">Integration Test</a>
        </li>
        <li>
          <a href="3.2.3-modularitytest.html">Modularity Test</a>
        </li>
        <li>
          <a href="3.2.4-overallintegration.html">Overall Integration for a Set of Traits</a>
        </li>
        <li>
          <a href="3.2.5-phylointegration.html">Phylogenetic Integration (phylo.int)</a>
        </li>
        <li>
          <a href="3.2.6-phylomodularity.html">Phylogenetic Modularity (phylo.mod)</a>
        </li>
        <li>
          <a href="3.2.7-globalintegration.html">Global Integration</a>
        </li>
        <li>
          <a href="3.2.8-compareCR.html">Comparison of Modularity (compare.CR)</a>
        </li>
        <li>
          <a href="3.2.9-comparepls.html">Comparison of Integration (compare.pls)</a>
        </li>
        <li>
          <a href="3.2.10-compareZVrel.html">Comparison of Overall Integration</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Phylogenetic Comparative Methods</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="3.1.1-ANOVARegression.html">Phylogenetic ANOVA/Regression (ProcD.pgls)</a>
        </li>
        <li>
          <a href="3.3.1-compareevolrates.html">Compare Evolutionary Rates (compare.evol.rates)</a>
        </li>
        <li>
          <a href="3.3.2-comparemultievol.html">Compare Multiple Evolutionary Rates (compare.multi.evol.rates)</a>
        </li>
        <li>
          <a href="3.2.5-phylointegration.html">Phylogenetic Morphological Integration (phylo.int)</a>
        </li>
        <li>
          <a href="3.2.6-phylomodularity.html">Phylogenetic Modularity (phylo.mod)</a>
        </li>
        <li>
          <a href="3.3.3-physignal.html">Assessing Phylogenetic Signal (physignal)</a>
        </li>
        <li>
          <a href="3.3.4-physignalz.html">Assessing Phylogenetic Signal Effect Size (physignal.z)</a>
        </li>
        <li>
          <a href="3.3.5-comparephysigz.html">Comparing Phylogenetic Signal Effect Sizes (compare.physignal.z)</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Visualizations
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="4.1-plotallspec.html">Plot Full Landmark Dataset (plotAllSpecimens)</a>
    </li>
    <li>
      <a href="4.2-plotreftarg.html">Plot Shape Differences (plotRefToTarget)</a>
    </li>
    <li>
      <a href="4.3-shapepred.html">Shape Prediction from Numeric Predictors (shape.predictor)</a>
    </li>
    <li>
      <a href="4.4-Ordplot.html">Ordination: PCA, pPCA, and PaCA Analyses (gm.prcomp)</a>
    </li>
    <li>
      <a href="4.5-plotallometry.html">Plot Allometry</a>
    </li>
    <li>
      <a href="4.6-plotspec.html">Plot 3D Specimen (plotspec)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Common Workflows
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="5.1-Group-Comparisons.html">Group Comparisons</a>
    </li>
    <li>
      <a href="4.2-VisShapeDiff.html">Visualizing Shape Differences</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore"><strong>Estimate Missing
Landmarks</strong></h1>
<h4 class="author">Dean Adams, Michael Collyer, Antigoni
Kaliontzopoulou, &amp; Mark A. Conaway</h4>

</div>


<p><br></p>
<div id="introduction" class="section level2">
<h2><strong>Introduction</strong></h2>
<p>Sometimes, specimens are not fully intact, and there are missing
landmarks. For these specimens, one cannot digitize the locations of
those points, as they are found in broken, or otherwise absent regions.
Because geometric morphometric methods requires objects containing a
complete set of landmarks for subsequent analyses (GPA superimposition
and downstream statistical assessments), one must do one of the
following three options:</p>
<ul>
<li>Eliminate all specimens with missing data</li>
<li>Eliminate all landmarks that are not found on all specimens</li>
<li>Estimate the locations of missing landmarks in some intelligent
manner</li>
</ul>
<p>Clearly, the first and second options are sub-optimal. Instead we
will estimate the locations of missing landmarks in
<code>geomorph</code>. There are three main ways that have been used to
estimate the positions of missing landmarks:</p>
<ol style="list-style-type: decimal">
<li>TPS interpolation</li>
<li>multivariate regression</li>
<li>exploiting symmetry</li>
</ol>
<p>Methods 1 &amp; 2 are implemented within the
<code>estimate.missing</code> function of <code>geomorph</code>, and the
third can be implemented using the R package Morpho <br></br></p>
<div
id="estimating-missing-data-using-the-estimate.missing-function-in-geomorph"
class="section level4">
<h4><strong>Estimating missing data using the
<code>estimate.missing</code> Function in
<code>geomorph</code></strong></h4>
<p>The <code>estimate.missing</code> function allows the user to
estimate the locations of missing landmarks using one of two methods; 1)
Thin-plate Spline, and 2) Regression. Here we will show how each of
these methods work, and the results that they generate.</p>
<div class="func">
<details>
<summary>
<font size="+2"> <code>estimate.missing()</code> </font>
</summary>
<p>This function allows for the estimation of missing landmark data in a
set of specimens. Each of the two possible methods requires a reference
in order to perform the estimation. The possible arguments, or input
values specified by the user, are below:</p>
<ul>
<li><span class="math inline">\(A\)</span>: An <a href="">array</a>
containing landmark coordinates for a set of specimens</li>
<li><span class="math inline">\(method\)</span>: Specifies the method
that should be used for estimating missing landmarks. The options are
“TPS”, or “Reg”. Each will be explained in detail below.</li>
</ul>
It is important to note that the “Reg” method requires a minimum of
<span class="math inline">\(2m+2\)</span> specimens in order to estimate
<span class="math inline">\(m\)</span> missing landmarks.
</details>
</div>
<p><br></br></p>
</div>
<div id="using-thin-plate-spline-interpolation-tps"
class="section level4">
<h4><strong>Using Thin-Plate Spline Interpolation
(<code>TPS</code>)</strong></h4>
<p>A more general approach is to use thin-plate spline interpolation.
Here, one uses the covariation among existing landmarks on one object to
infer the position of the missing landmark on the other.</p>
<p>Let’s see an example using the salamander data including with
<code>geomorph</code>: <br></br></p>
<div class="code">
<pre class="r custom-inline"><code>data(&quot;plethodon&quot;)
data(&quot;plethspecies&quot;)
GPA &lt;- gpagen(plethodon$land)</code></pre>
</div>
<p><br></br></p>
<p><img src="1.3.7-EstMissing_files/figure-html/PCAandplot-1.png" width="70%" style="display: block; margin: auto;" /></p>
<p>The above figure illustrates the extremes of variation of this
dataset (bottom left and right),as well as its mean shape (top-right).
The <code>estimate.missing</code> function will use this mean shape as
reference. Incomplete specimens will be aligned to the reference, and
the locations of missing landmarks will be estimated using a thin-plate
spline.</p>
<p>Now let’s delete some landmarks from the three specimens highlighted
in red:</p>
<p><img src="1.3.7-EstMissing_files/figure-html/deletelmkspleth-1.png" width="70%" style="display: block; margin: auto;" /></p>
<p><br></br></p>
<p>Finally, let’s estimate the landmarks via the thin-plate spline and
compare to the actual specimens:</p>
<div class="code">
<pre class="r custom-inline"><code>new.tps&lt;-estimate.missing(shapes.missing,method=&quot;TPS&quot;)</code></pre>
</div>
<p><br></br></p>
<p>Now plot and compare:</p>
<p><img src="1.3.7-EstMissing_files/figure-html/plotcomparepleth-1.png" width="90%" style="display: block; margin: auto;" /></p>
<pre class="custom-inline"><code>##            1          2          3
## 1 0.00000000 0.07704341 0.08109931
## 2 0.07704341 0.00000000 0.05623248
## 3 0.08109931 0.05623248 0.00000000</code></pre>
<p>The numbers above are measures of shape difference (Procrustes
distance). As we can see, this is pretty good also, and does not require
object symmetry. <br></br></p>
</div>
<div id="using-regression-reg" class="section level4">
<h4><strong>Using Regression (<code>Reg</code>)</strong></h4>
<p>Perhaps the most general approach is to use regression to estimate
the locations of missing landmarks. Using the same salamander data:</p>
<div class="code">
<pre class="r custom-inline"><code>new.reg&lt;-estimate.missing(shapes.missing,method=&quot;Reg&quot;)
new.reg&lt;-gpagen(new.reg, print.progress=FALSE)$coords</code></pre>
</div>
<p><br></br></p>
<p>Now plot and compare:</p>
<div class="code">
<pre class="r custom-inline"><code>par(mfrow=c(2,2))
plotRefToTarget(ref,ref,links=plethodon$links)
mtext(&quot;Reference&quot;)
plotRefToTarget(ref,GPA$coords[,,1],links=plethodon$links)
mtext(&quot;TPS: Reference --&gt; Original&quot;)
plotRefToTarget(ref,new.reg[,,1],links=plethodon$links)
mtext(&quot;TPS: Ref --&gt; Estimated Specimen&quot;)
plotRefToTarget(GPA$coords[,,1],new.reg[,,1],links=plethodon$links)
mtext(&quot;TPS: Original --&gt; Estimated Specimen&quot;)</code></pre>
</div>
<p><br></br></p>
<p><img src="1.3.7-EstMissing_files/figure-html/plotcomparereg1-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="code">
<pre class="r custom-inline"><code>as.matrix(dist(rbind(t(matrix(ref)),t(matrix(GPA$coords[,,1])),t(matrix(new.tps[,,1])))))</code></pre>
<pre class="custom-inline"><code>##            1          2          3
## 1 0.00000000 0.07704341 0.08109931
## 2 0.07704341 0.00000000 0.05623248
## 3 0.08109931 0.05623248 0.00000000</code></pre>
</div>
<p>As we can see from the Procrustes distances, this method is even
better at estimating the locations of missing landmarks!</p>
<p><br></br></p>
</div>
<div id="exploiting-object-symmetry" class="section level4">
<h4><strong>Exploiting Object Symmetry</strong></h4>
<p>If one’s specimens are symmetric, the locations on one side of the
axis of symmetry may be used to estimate the locations of missing
landmarks on the other side of the symmetry axis. Below is a simple
example (<strong>NOTE: we will use the R-package <code>Morpho</code> for
the symmetry-based landmark estimation</strong>).</p>
<p><br></p>
<p>Here is some sample data (lizard skulls)</p>
<div class="code">
<pre class="r custom-inline"><code>#read data
mydat&lt;-readland.tps(&quot;./Data-figures/LizardShape.tps&quot;,warnmsg = FALSE)
inc.shape &lt;- gpagen(mydat,print.progress = FALSE)$coords
mydat &lt;- gpagen(mydat, print.progress = FALSE)$coords

#landmark pairing
land.prs&lt;-matrix(c(13, 14,2,11,15, 16,27, 28,3,10,17,19,4, 9,20, 21,22, 24,25, 26,5, 8, 6 ,7),ncol=2,byrow=T)

#plotPCA
links&lt;-read.table(&quot;./Data-figures/lizardlinks.txt&quot;,header=T)
ref&lt;-mshape(mydat)
shape.pca &lt;- gm.prcomp(mydat)
col.spec &lt;- rep(&quot;black&quot;,dim(mydat)[[3]]); col.spec[62] =&quot;red&quot;
sz.spec &lt;- rep(1.0,dim(mydat)[[3]]); sz.spec[62] =2</code></pre>
</div>
<p><br></br></p>
<p><img src="1.3.7-EstMissing_files/figure-html/plotallspecimens-1.png" width="70%" style="display: block; margin: auto;" />
<br></br></p>
<p>In the right-hand plot, the red dot represents the location of the
specimen (#62) for which we have missing data.</p>
<p>In the above code, <code>land.prs</code> is a matrix identifying the
bilateral pairs of landmarks in this dataset. <br></br></p>
<div class="code">
<pre class="r custom-inline"><code>land.prs</code></pre>
<pre class="custom-inline"><code>##       [,1] [,2]
##  [1,]   13   14
##  [2,]    2   11
##  [3,]   15   16
##  [4,]   27   28
##  [5,]    3   10
##  [6,]   17   19
##  [7,]    4    9
##  [8,]   20   21
##  [9,]   22   24
## [10,]   25   26
## [11,]    5    8
## [12,]    6    7</code></pre>
</div>
<p><br></br></p>
<p>Here we show the missing landmarks:</p>
<p><img src="Data-figures/missing/LizardMissing.png" width="50%" style="display: block; margin: auto;" />
<br></br></p>
<pre class="r custom-inline"><code>inc.shape[,,62]</code></pre>
<pre class="custom-inline"><code>##               X             Y
## 1   0.358795641  1.250158e-04
## 2   0.168067228  8.877281e-02
## 3   0.032826986  4.511770e-02
## 4            NA            NA
## 5            NA            NA
## 6  -0.251094873  3.265606e-02
## 7  -0.243365354 -3.247129e-02
## 8  -0.237993695 -1.154278e-01
## 9  -0.065715500 -1.382644e-01
## 10  0.029898661 -4.652521e-02
## 11  0.167430767 -8.958667e-02
## 12  0.319641445  3.504307e-03
## 13  0.259497241  5.861418e-02
## 14           NA            NA
## 15  0.144292265  6.447751e-02
## 16  0.145964948 -6.813254e-02
## 17  0.011555678  4.339545e-02
## 18 -0.010393806 -4.891806e-03
## 19  0.008404225 -4.346222e-02
## 20 -0.062401338  1.039209e-01
## 21 -0.059651939 -1.068104e-01
## 22 -0.129421710  2.279160e-02
## 23 -0.101289883 -6.032779e-06
## 24 -0.133789681 -2.688542e-02
## 25 -0.192467410  1.208056e-02
## 26 -0.190946529 -1.123231e-02
## 27  0.036363446  1.251641e-01
## 28           NA            NA</code></pre>
<p><br></br></p>
</div>
<div id="estimating-missing-data-by-exploiting-object-symmetry"
class="section level4">
<h4><strong>Estimating missing data by exploiting object
symmetry</strong></h4>
<p>Now let’s estimate the locations of the missing landmarks by their
respective landmarks on the other side of the symmetry axis. Exploiting
symmetry is a biologically-driven manner of estimating missing
landmarks, but is only appropriate for symmetric objects.This is most
easily accomplished using the <code>fixLMmirror</code> function found in
the package <code>Morpho</code>. All that is required is a <a
href="Tutorials/1.1-ReadingData.html">3D array</a> of <a
href="Tutorials/2.1-GeneralizedProcrustesAnalysis.html">Procrustes-aligned</a>
landmark coordinates with missing data (<code>inc.shape</code>), and a
2-column matrix identifying paired landmark numbers
(<code>land.prs</code>) <br></br></p>
<div class="func">
<details>
<summary>
<font size="+2"> <code>fixLMmirror() (Expand for more details)</code>
</font>
</summary>
<p>This function allows you to extract landmark coordinates from a
“shapes,” file in your working environment. The possible arguments, or
input values specified by the user, are below:</p>
<ul>
<li><span class="math inline">\(x\)</span>: A matrix or an array of
landmark coordinates. These can be either 2D or 3D</li>
<li><span class="math inline">\(pairedLM\)</span>: A matrix identifying
pairs of landmarks (as shown above)</li>
</ul>
</details>
</div>
<p><br></br></p>
<div class="code">
<pre class="r custom-inline"><code>inc.shape &lt;- fixLMmirror(inc.shape, land.prs)</code></pre>
</div>
<p><br></br></p>
<p>Now plot and compare via Procrustes distance with the original:</p>
<p><img src="1.3.7-EstMissing_files/figure-html/plotcompareliz1-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="code">
<pre class="r custom-inline"><code>par(mfrow=c(1,1)) 
dist(rbind(t(matrix(inc.shape[,,62])),t(matrix(mydat[,,62]))))</code></pre>
<pre class="custom-inline"><code>##            1
## 2 0.02107804</code></pre>
</div>
<p><br></br></p>
<p>The number above is the amount of shape difference expressed as
Procrustes distance. <br></br></p>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
