<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Dean Adams, Michael Collyer &amp; Antigoni Kaliontzopoulou" />


<title>Visualizing Shape Differences and Principal Components Analysis</title>

<script src="site_libs/header-attrs-2.19/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="style/tutorial_theme.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"><img id="logo"style="height: 25px;" src="geomorph.png" /></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    GM Preliminaries
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="1.1-GeomorphDataTypes.html">Geomorph Data Types</a>
    </li>
    <li>
      <a href="1.2-ReadingData.html">Reading Data</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Data Manipulation</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="1.3.1-GeomorphDataFrame.html">Geomorph Data Frame</a>
        </li>
        <li>
          <a href="1.3.2-CombineLmkConfig.html">Combine Landmark Configurations (combine.subsets)</a>
        </li>
        <li>
          <a href="1.3.3-Interlmkdist.html">Interlandmark Distances (interlmkdist)</a>
        </li>
        <li>
          <a href="1.3.4-Flip-RotateLmks.html">Rotate or Flip Landmark Configurations (rotate.coords)</a>
        </li>
        <li>
          <a href="1.3.5-AllAboutArrays.html">Create 3D Array (arrayspecs)</a>
        </li>
        <li>
          <a href="1.3.6-Two.d.array.html">Convert Array to Matrix (two.d.array)</a>
        </li>
        <li>
          <a href="1.3.7-EstMissing.html">Estimating Missing Landmarks</a>
        </li>
        <li>
          <a href="1.3.8-IdentifyOutliers.html">Outlier Identification</a>
        </li>
        <li>
          <a href="1.3.9-StandardizeArticulations.html">Standardizing Articulations</a>
        </li>
        <li>
          <a href="1.3.10-SymmetricShapeGeneration.html">Symmetric Shape Generation</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">R Data Basics</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="1.4.1-SpecimenNames.html">Adding &amp; Changing Specimen Names</a>
        </li>
        <li>
          <a href="1.4.2-ExtractingClassifiers.html">Extracting Classifiers</a>
        </li>
        <li class="dropdown-header">Reflecting Landmarks</li>
        <li>
          <a href="1.4.3-SwappingLandmarks.html">Swapping Landmarks</a>
        </li>
        <li>
          <a href="1.4.4-DeletingSpecLmk.html">Delete Specs. and Landmarks</a>
        </li>
        <li>
          <a href="1.4.5-MatchSpectoPhy.html">Matching Data to a Phylogeny</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Procrustes Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="2.1-GPAIntro.html">GPA Introduction</a>
    </li>
    <li>
      <a href="2.2-GPAfixedlmks.html">GPA with fixed landmarks</a>
    </li>
    <li>
      <a href="2.3-GPAcurves.html">GPA with curve semilandmarks</a>
    </li>
    <li>
      <a href="2.4-GPAsurface.html">GPA with surface semilandmarks</a>
    </li>
    <li class="dropdown-header">gpagen Advanced Options</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analytics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Linear Models</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="3.1.1-ANOVARegression.html">ANOVA/Regression (ProcD.lm)</a>
        </li>
        <li>
          <a href="3.1.2-Phylo-ANOVA-Regression.html">Phylogenetic ANOVA/Regression (ProcD.pgls)</a>
        </li>
        <li>
          <a href="3.1.3-Trajectoryanalysis.html">Phenotypic Trajectory Analysis</a>
        </li>
        <li>
          <a href="3.1.4-Pairwisecomp.html">Pairwise Comparisons (pairwise)</a>
        </li>
        <li>
          <a href="3.1.5-TwoBPLS.html">Two-block Partial Least Squares (two.b.pls)</a>
        </li>
        <li>
          <a href="3.1.6-Bilatsymmanalysis.html">Analysis of Bilateral Symmetry (bilat.symmetry)</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Shape Patterns</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="3.2.1-twobpls.html">Two-block Partial Least Squares (two.b.pls)</a>
        </li>
        <li>
          <a href="3.2.2-integrationtest.html">Integration Test</a>
        </li>
        <li>
          <a href="3.2.3-modularitytest.html">Modularity Test</a>
        </li>
        <li>
          <a href="3.2.4-overallintegration.html">Overall Integration for a Set of Traits</a>
        </li>
        <li>
          <a href="3.2.5-phylointegration.html">Phylogenetic Integration (phylo.int)</a>
        </li>
        <li>
          <a href="3.2.6-phylomodularity.html">Phylogenetic Modularity (phylo.mod)</a>
        </li>
        <li>
          <a href="3.2.7-globalintegration.html">Global Integration</a>
        </li>
        <li>
          <a href="3.2.8-compareCR.html">Comparison of Modularity (compare.CR)</a>
        </li>
        <li>
          <a href="3.2.9-comparepls.html">Comparison of Integration (compare.pls)</a>
        </li>
        <li>
          <a href="3.2.10-compareZVrel.html">Comparison of Overall Integration</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Phylogenetic Comparative Methods</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="3.1.1-ANOVARegression.html">Phylogenetic ANOVA/Regression (ProcD.pgls)</a>
        </li>
        <li>
          <a href="3.3.1-compareevolrates.html">Compare Evolutionary Rates (compare.evol.rates)</a>
        </li>
        <li>
          <a href="3.3.2-comparemultievol.html">Compare Multiple Evolutionary Rates (compare.multi.evol.rates)</a>
        </li>
        <li>
          <a href="3.2.5-phylointegration.html">Phylogenetic Morphological Integration (phylo.int)</a>
        </li>
        <li>
          <a href="3.2.6-phylomodularity.html">Phylogenetic Modularity (phylo.mod)</a>
        </li>
        <li>
          <a href="3.3.3-physignal.html">Assessing Phylogenetic Signal (physignal)</a>
        </li>
        <li>
          <a href="3.3.4-physignalz.html">Assessing Phylogenetic Signal Effect Size (physignal.z)</a>
        </li>
        <li>
          <a href="3.3.5-comparephysigz.html">Comparing Phylogenetic Signal Effect Sizes (compare.physignal.z)</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Visualizations
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="4.1-plotallspec.html">Plot Full Landmark Dataset (plotAllSpecimens)</a>
    </li>
    <li>
      <a href="4.2-plotreftarg.html">Plot Shape Differences (plotRefToTarget)</a>
    </li>
    <li>
      <a href="4.3-shapepred.html">Shape Prediction from Numeric Predictors (shape.predictor)</a>
    </li>
    <li>
      <a href="4.4-Ordplot.html">Ordination: PCA, pPCA, and PaCA Analyses (gm.prcomp)</a>
    </li>
    <li>
      <a href="4.5-plotallometry.html">Plot Allometry</a>
    </li>
    <li>
      <a href="4.6-plotspec.html">Plot 3D Specimen (plotspec)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Common Workflows
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="5.1-Group-Comparisons.html">Group Comparisons</a>
    </li>
    <li>
      <a href="4.2-VisShapeDiff.html">Visualizing Shape Differences</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore"><strong>Visualizing Shape Differences and
Principal Components Analysis</strong></h1>
<h4 class="author">Dean Adams, Michael Collyer &amp; Antigoni
Kaliontzopoulou</h4>

</div>


<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>In this tutorial, we demonstrate some of the graphical capabilities
of <code>geomorph</code> that may be used to visualize patterns of shape
variation, and interpret particular shape differences between specimens.
These types of plots, shape difference plots and principal components
plots, are found in nearly every empirical publication using geometric
morphometric shape data, and are thus of primary importance.
<code>Geomorph</code> now has considerable flexibility in plot options
for these visualizations, and we encourage you to explore these options
and consult the help pages of the various functions.</p>
<div id="visualizing-shape-differences-using-geomorph"
class="section level3 tabset tabset-fade tabset-pills">
<h3 class="tabset tabset-fade tabset-pills"><strong>Visualizing shape
differences using geomorph</strong></h3>
<div id="preliminaries" class="section level4">
<h4>0: Preliminaries</h4>
<p>First, let´s load some example data and plot the specimens (before
and after GPA alignment):</p>
<pre class="r custom-inline"><code># library(geomorph)
# data(plethodon)
# Y.gpa &lt;- gpagen(plethodon$land, print.progress = F)    # GPA-alignment
# 
# par(mfrow=c(1,2)) 
# plotAllSpecimens(plethodon$land, links=plethodon$links)  # Raw data
# mtext(&quot;Raw Data&quot;)
# plotAllSpecimens(Y.gpa$coords, links=plethodon$links)    # GPA-aligned data
# mtext(&quot;GPA-Aligned Specimens&quot;)</code></pre>
<pre class="r custom-inline"><code># par(mfrow=c(1,1)) </code></pre>
</div>
<div id="visualize-shape-differences-between-specimens"
class="section level4">
<h4>1: Visualize Shape Differences Between Specimens</h4>
<p>Visualizing shape deformations is <strong>ALWAYS</strong> relative to
some reference. That is, we provide graphical depictions of shape
<em>differences</em> <span
class="math inline">\(\small{Reference\rightarrow{Target}}\)</span>.
Typically, we use the overall average as a reference, and deform it into
some target using the function <code>plotRefToTarget</code>. There are
quite a few plotting options available: a few of them are illustrated
here (but see the help files for more details!).</p>
<p>**Here we demonstrate several shape deformation plots as found in
<code>geomorph</code></p>
<p>Deformations Specimen to Specimen</p>
<pre class="r custom-inline"><code># ref &lt;- mshape(Y.gpa$coords)
# 
# plotRefToTarget(ref,Y.gpa$coords[,,39], links=plethodon$links)
# mtext(&quot;TPS&quot;)
# plotRefToTarget(ref,Y.gpa$coords[,,39], mag=2.5, links=plethodon$links)
# mtext(&quot;TPS: 2.5X magnification&quot;)
# par(mfrow=c(1,1))
# 
# plotRefToTarget(ref,Y.gpa$coords[,,39], links=plethodon$links, method=&quot;vector&quot;, mag=3)
# mtext(&quot;Vector Displacements&quot;)
# plotRefToTarget(ref,Y.gpa$coords[,,39], links=plethodon$links,gridPars=gridPar(pt.bg=&quot;red&quot;, link.col=&quot;green&quot;, pt.size = 1), method=&quot;vector&quot;, mag=3)
# mtext(&quot;Vector Displacements: Other Options&quot;)
# 
# plotRefToTarget(ref,Y.gpa$coords[,,39], mag=2, outline=plethodon$outline)  
# mtext(&quot;Outline Deformation&quot;)
# plotRefToTarget(ref,Y.gpa$coords[,,39], method=&quot;points&quot;, outline=plethodon$outline)
# mtext(&quot;Outline Deformations Ref (gray) &amp; and Tar (black)&quot;)</code></pre>
</div>
<div id="advanced-shape-plotting-shape-prediction"
class="section level4">
<h4>2: Advanced Shape Plotting: Shape Prediction</h4>
<p>Many times, we wish to visualize shape deformations based on a
particular statistical model. For instance, we may wish to visualize
group differences from ANOVA, or visualize how shape changes along some
axis, like a regression line or a PC-axis. All of these require
<em>Predicting</em> shapes based on some mathematical model, and then
visualizing shape deformations using those predicted shapes.</p>
<p>While the mathematics of this can become rather involved, it is
straightforward. Fortunately, <code>geomorph</code> has a wonderful and
general function to accomplish this task: <code>shape.predictor</code>.
Here we demonstrate a few of its capabilities (**NOTE: in future
lectures we will discuss the mathematics of these statistical methods.
Here the visualization tools only are demonstrated):</p>
<p>PCA Shape Predictions</p>
<pre class="r custom-inline"><code># M &lt;- mshape(Y.gpa$coords)
# PCA &lt;- plotTangentSpace(Y.gpa$coords)
# PC &lt;- PCA$pc.scores[,1]
# preds &lt;- shape.predictor(Y.gpa$coords, x= PC, Intercept = FALSE, 
#                          pred1 = min(PC), pred2 = max(PC)) # PC 1 extremes, more technically
# plotRefToTarget(M, preds$pred1, links = plethodon$links)
# mtext(&quot;PC1 - Min.&quot;)
# plotRefToTarget(M, preds$pred2, links = plethodon$links)
# mtext(&quot;PC1 - Max.&quot;)</code></pre>
<p>Shape Predictions Along a Regression</p>
<p>Many times, we fit a linear model in the form of regression (e.g.,
allometry). Here is how we visualize shape changes along the extremes of
that regression using <code>shape.predictor</code>:</p>
<pre class="r custom-inline"><code># gdf &lt;- geomorph.data.frame(Y.gpa)
# plethAllometry &lt;- procD.lm(coords ~ log(Csize), data=gdf, print.progress = FALSE)
# allom.plot &lt;- plot(plethAllometry, 
#                    type = &quot;regression&quot;, 
#                    predictor = log(gdf$Csize),
#                    reg.type =&quot;PredLine&quot;) # make sure to have a predictor 
# 
# preds &lt;- shape.predictor(plethAllometry$GM$fitted, x= allom.plot$RegScore, Intercept = FALSE, 
#                          predmin = min(allom.plot$RegScore), 
#                          predmax = max(allom.plot$RegScore)) 
# plotRefToTarget(M, preds$predmin, mag=3, links = plethodon$links)
# plotRefToTarget(M, preds$predmax, mag=3, links = plethodon$links)</code></pre>
<p>Shape Predictions of Group Differences</p>
<p>Another very common statistical design is one of evaluating group
differences in shape (i.e., ANOVA). Here we wish to visualize shape
differences among groups. Below is an example:</p>
<pre class="r custom-inline"><code># gdf &lt;- geomorph.data.frame(Y.gpa, species = plethodon$species, site = plethodon$site)
# pleth.anova &lt;- procD.lm(coords ~ species*site, data=gdf, print.progress = FALSE)
# X &lt;- pleth.anova$X
# X # includes intercept; remove for better functioning 
# X &lt;- X[,-1]
# symJord &lt;- c(0,1,0) # design for P. Jordani in sympatry
# alloJord &lt;- c(0,0,0) # design for P. Jordani in allopatry
# preds &lt;- shape.predictor(arrayspecs(pleth.anova$fitted, 12, 2), x = X, Intercept = TRUE, 
#                          symJord=symJord, alloJord=alloJord)
# plotRefToTarget(M, preds$symJord, links = plethodon$links, mag=2)
# plotRefToTarget(M, preds$alloJord, links = plethodon$links, mag=2)</code></pre>
</div>
<div id="pca-plot-tangent-space" class="section level4">
<h4>3: PCA Plot (Tangent Space)</h4>
<p>In <code>geomorph</code>, there are two ways to obtain a principal
components plot of shape variation (i.e, a visualization of Tangent
Space). Here is the first option, using
<code>plotTangentSpace</code>:</p>
<pre class="r custom-inline"><code># plotTangentSpace(Y.gpa$coords, groups = interaction(plethodon$species, plethodon$site))</code></pre>
<p>Alternatively, one could use the function <code>gm.prcomp</code>.
This has additional flexibility as compared to the previous function,
some of which we show here (later we will show phylogenetic PCA
plots):</p>
<pre class="r custom-inline"><code># pleth.raw &lt;- gm.prcomp(Y.gpa$coords)
# 
# gps &lt;- as.factor(paste(plethodon$species, plethodon$site))
# plot(pleth.raw)
# par(mar=c(2, 2, 2, 2))
# plot(pleth.raw, pch=22, cex = 1.5, bg = gps) 
# #  Add things as desired using standard R plotting
# text(par()$usr[1], 0.1*par()$usr[3], labels = &quot;PC1 - 36.74%&quot;, pos = 4, font = 2)
# text(0, 0.95*par()$usr[4], labels = &quot;PC2 - 31.02%&quot;, pos = 4, font = 2)
# legend(&quot;topleft&quot;, pch=22, pt.bg = unique(gps), legend = levels(gps))</code></pre>
</div>
<div id="selecting-shapes-to-visualize-in-real-time"
class="section level4">
<h4>4: Selecting Shapes to Visualize in Real Time</h4>
<p>One recent enhancement of <code>geomorph</code> is the
<code>picknplot.shape</code> function. With this function, one can
select locations in a statistical dataspace, and <code>geomorph</code>
will mathematically back-translate the statistical summary values in the
plot to obtain landmark coordinates, and will then generate a thin-plate
spline deformation grid of a specimen that would exist at that location
in the dataspace.</p>
<p>This real-time visualization is especially useful for viewing regions
of morphospace that are unoccupied by the sample of specimens, and for
exploring other non-sampled shapes.</p>
<div
id="note-because-this-is-an-interactive-function-the-code-is-repeated-below-but-not-run."
class="section level5">
<h5>Note: because this is an interactive function, the code is repeated
below but not run.</h5>
<pre class="r custom-inline"><code># 2d
# data(plethodon) 
# Y.gpa &lt;- gpagen(plethodon$land)
# pleth.pca &lt;- gm.prcomp(Y.gpa$coords)
# pleth.pca.plot &lt;- plot(pleth.pca)
# picknplot.shape(pleth.pca.plot) 
# May change arguments for plotRefToTarget
# picknplot.shape(plot(pleth.pca), method = &quot;points&quot;, mag = 3, links=plethodon$links)</code></pre>
</div>
</div>
<div id="d-warping" class="section level4">
<h4>5: 3D warping</h4>
<p><code>Geomorph</code> also has the ability to generate shape
deformations of 3D objects. Again this is interactive code, so it is
repeated, but not run, below:</p>
<pre class="r custom-inline"><code>#scallops &lt;- readland.tps(&quot;Data/scallops for viz.tps&quot;, specID = &quot;ID&quot;)
#ref &lt;- mshape(scallops)
#refmesh &lt;- warpRefMesh(read.ply(&quot;Data/glyp02L.ply&quot;), 
#                       scallops[,,1], ref, color=NULL, centered=T)
#plot.gm.prcomp(scallops, axis1 = 1, axis2 = 2, warpgrids=T, mesh= refmesh)</code></pre>
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
