<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mark A. Conaway" />


<title>Shape Prediction from Numeric Predictors</title>

<script src="site_libs/header-attrs-2.19/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="style/tutorial_theme.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"><img id="logo"style="height: 25px;" src="geomorph.png" /></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    GM Preliminaries
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="1.1-GeomorphDataTypes.html">Geomorph Data Types</a>
    </li>
    <li>
      <a href="1.2-ReadingData.html">Reading Data</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Data Manipulation</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="1.3.1-GeomorphDataFrame.html">Geomorph Data Frame</a>
        </li>
        <li>
          <a href="1.3.2-CombineLmkConfig.html">Combine Landmark Configurations (combine.subsets)</a>
        </li>
        <li>
          <a href="1.3.3-Interlmkdist.html">Interlandmark Distances (interlmkdist)</a>
        </li>
        <li>
          <a href="1.3.4-Flip-RotateLmks.html">Rotate or Flip Landmark Configurations (rotate.coords)</a>
        </li>
        <li>
          <a href="1.3.5-AllAboutArrays.html">Create 3D Array (arrayspecs)</a>
        </li>
        <li>
          <a href="1.3.6-Two.d.array.html">Convert Array to Matrix (two.d.array)</a>
        </li>
        <li>
          <a href="1.3.7-EstMissing.html">Estimating Missing Landmarks</a>
        </li>
        <li>
          <a href="1.3.8-IdentifyOutliers.html">Outlier Identification</a>
        </li>
        <li>
          <a href="1.3.9-StandardizeArticulations.html">Standardizing Articulations</a>
        </li>
        <li>
          <a href="1.3.10-SymmetricShapeGeneration.html">Symmetric Shape Generation</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">R Data Basics</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="1.4.1-SpecimenNames.html">Adding &amp; Changing Specimen Names</a>
        </li>
        <li>
          <a href="1.4.2-ExtractingClassifiers.html">Extracting Classifiers</a>
        </li>
        <li class="dropdown-header">Reflecting Landmarks</li>
        <li>
          <a href="1.4.3-SwappingLandmarks.html">Swapping Landmarks</a>
        </li>
        <li>
          <a href="1.4.4-DeletingSpecLmk.html">Delete Specs. and Landmarks</a>
        </li>
        <li>
          <a href="1.4.5-MatchSpectoPhy.html">Matching Data to a Phylogeny</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Procrustes Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="2.1-GPAIntro.html">GPA Introduction</a>
    </li>
    <li>
      <a href="2.2-GPAfixedlmks.html">GPA with fixed landmarks</a>
    </li>
    <li>
      <a href="2.3-GPAcurves.html">GPA with curve semilandmarks</a>
    </li>
    <li>
      <a href="2.4-GPAsurface.html">GPA with surface semilandmarks</a>
    </li>
    <li class="dropdown-header">gpagen Advanced Options</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analytics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Linear Models</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="3.1.1-ANOVARegression.html">ANOVA/Regression (ProcD.lm)</a>
        </li>
        <li>
          <a href="3.1.2-Phylo-ANOVA-Regression.html">Phylogenetic ANOVA/Regression (ProcD.pgls)</a>
        </li>
        <li>
          <a href="3.1.3-Trajectoryanalysis.html">Phenotypic Trajectory Analysis</a>
        </li>
        <li>
          <a href="3.1.4-Pairwisecomp.html">Pairwise Comparisons (pairwise)</a>
        </li>
        <li>
          <a href="3.1.5-TwoBPLS.html">Two-block Partial Least Squares (two.b.pls)</a>
        </li>
        <li>
          <a href="3.1.6-Bilatsymmanalysis.html">Analysis of Bilateral Symmetry (bilat.symmetry)</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Shape Patterns</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="3.2.1-twobpls.html">Two-block Partial Least Squares (two.b.pls)</a>
        </li>
        <li>
          <a href="3.2.2-integrationtest.html">Integration Test</a>
        </li>
        <li>
          <a href="3.2.3-modularitytest.html">Modularity Test</a>
        </li>
        <li>
          <a href="3.2.4-overallintegration.html">Overall Integration for a Set of Traits</a>
        </li>
        <li>
          <a href="3.2.5-phylointegration.html">Phylogenetic Integration (phylo.int)</a>
        </li>
        <li>
          <a href="3.2.6-phylomodularity.html">Phylogenetic Modularity (phylo.mod)</a>
        </li>
        <li>
          <a href="3.2.7-globalintegration.html">Global Integration</a>
        </li>
        <li>
          <a href="3.2.8-compareCR.html">Comparison of Modularity (compare.CR)</a>
        </li>
        <li>
          <a href="3.2.9-comparepls.html">Comparison of Integration (compare.pls)</a>
        </li>
        <li>
          <a href="3.2.10-compareZVrel.html">Comparison of Overall Integration</a>
        </li>
      </ul>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Phylogenetic Comparative Methods</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="3.1.1-ANOVARegression.html">Phylogenetic ANOVA/Regression (ProcD.pgls)</a>
        </li>
        <li>
          <a href="3.3.1-compareevolrates.html">Compare Evolutionary Rates (compare.evol.rates)</a>
        </li>
        <li>
          <a href="3.3.2-comparemultievol.html">Compare Multiple Evolutionary Rates (compare.multi.evol.rates)</a>
        </li>
        <li>
          <a href="3.2.5-phylointegration.html">Phylogenetic Morphological Integration (phylo.int)</a>
        </li>
        <li>
          <a href="3.2.6-phylomodularity.html">Phylogenetic Modularity (phylo.mod)</a>
        </li>
        <li>
          <a href="3.3.3-physignal.html">Assessing Phylogenetic Signal (physignal)</a>
        </li>
        <li>
          <a href="3.3.4-physignalz.html">Assessing Phylogenetic Signal Effect Size (physignal.z)</a>
        </li>
        <li>
          <a href="3.3.5-comparephysigz.html">Comparing Phylogenetic Signal Effect Sizes (compare.physignal.z)</a>
        </li>
      </ul>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Visualizations
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="4.1-plotallspec.html">Plot Full Landmark Dataset (plotAllSpecimens)</a>
    </li>
    <li>
      <a href="4.2-plotreftarg.html">Plot Shape Differences (plotRefToTarget)</a>
    </li>
    <li>
      <a href="4.3-shapepred.html">Shape Prediction from Numeric Predictors (shape.predictor)</a>
    </li>
    <li>
      <a href="4.4-Ordplot.html">Ordination: PCA, pPCA, and PaCA Analyses (gm.prcomp)</a>
    </li>
    <li>
      <a href="4.5-plotallometry.html">Plot Allometry</a>
    </li>
    <li>
      <a href="4.6-plotspec.html">Plot 3D Specimen (plotspec)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Common Workflows
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="5.1-Group-Comparisons.html">Group Comparisons</a>
    </li>
    <li>
      <a href="4.2-VisShapeDiff.html">Visualizing Shape Differences</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore"><strong>Shape Prediction from Numeric
Predictors</strong></h1>
<h4 class="author">Mark A. Conaway</h4>

</div>


<div id="introduction" class="section level2">
<h2><strong>Introduction</strong></h2>
<p>Given one or more potential linear predictors, it is an extremely
useful tool in geometric morphometrics to be able to visualize the
effect that a predictor has on the shape of a specimen. The
<code>shape.predictor</code> function in geomorph allows the user to do
this using linear predictors such as <a href="">PC scores</a>,
allometric relationships, or <a href="">partial least squares</a>
regression.</p>
<div class="func">
<details>
<summary>
<font size="+2"> <code>shape.predictor()</code> </font>
</summary>
<ul>
<li><span class="math inline">\(A\)</span>: A 3D array (p x k x n)
containing Procrustes-aligned shape variables.</li>
<li><span class="math inline">\(x\)</span>: A vector or matrix (or a
list containing either of these) of linear (numeric) predictors. See
tutorials on <a href="">Ordination methods</a> for instructions on how
to generate these.</li>
<li><span class="math inline">\(Intercept\)</span>: Logical value (TRUE
or FALSE) to indicate whether an intercept should be used in the linear
equation for predictions. Generally, this value will be FALSE for shape
predictions made in ordination plots. It should be TRUE in cases where
the expected shape at the point the predictor has a value of 0 is not
the mean shape.</li>
<li><span class="math inline">\(method\)</span>: A choice between least
squares (LS) or partial least squares (PLS) regression for prediction.
The function defaults to LS prediction. PLS might be chosen in cases
where correlation is preferred over linear regression. If PLS is chosen,
a <a href="">two-block PLS analysis</a> using <code>two.b.pls</code>
should be performed first, as only the first singular vector for
predictors will be used for defining prediction criteria (see
below).</li>
</ul>
</details>
</div>
<p><br></br></p>
<div id="numeric-predictors"
class="section level3 tabset tabset-fade tabset-pills">
<h3 class="tabset tabset-fade tabset-pills"><strong>Numeric
Predictors</strong></h3>
<p><br></br></p>
<div id="shape-prediction-using-pc-scores" class="section level4">
<h4>1: Shape Prediction Using PC scores</h4>
<p>To predict shape changes based on PC scores, one must have results of
a <a href="">PCA</a> using the <code>gm.prcomp</code> function. For the
purposes of this example, we will assume the reader has done this
already. The next step is to extract the predictors (PCs) for use in the
function:</p>
<pre class="r custom-inline"><code>PC &lt;- PCA$x[,1]
PC</code></pre>
<pre class="custom-inline"><code>##             1             2             3             4             5             6             7 
## -0.0369930887 -0.0007493689  0.0056004751 -0.0134808326 -0.0334696064 -0.0052144953 -0.0030000192 
##             8             9            10            11            12            13            14 
## -0.0132974364 -0.0309852193 -0.0219654094 -0.0568125309 -0.0685392442 -0.0628850421 -0.0913269966 
##            15            16            17            18            19            20            21 
## -0.0623752317 -0.0495302423 -0.0404228370 -0.0499361411 -0.0691803602 -0.0475317885  0.0401310138 
##            22            23            24            25            26            27            28 
##  0.0517139033  0.0330882584  0.0433375155  0.0355450979  0.0215592389  0.0497210635  0.0286722364 
##            29            30            31            32            33            34            35 
##  0.0513588907  0.0338626707  0.0553310849  0.0322009051  0.0378707876  0.0373195281  0.0369264346 
##            36            37            38            39            40 
##  0.0348882260  0.0464477550  0.0340835994  0.0134507319  0.0345864739</code></pre>
<p>This returns the 40 PCs for the first specimen in our dataset. This
could be expanded to multiple specimens within (apparent) groups. Now we
put them to use in the function:</p>
<pre class="r custom-inline"><code>preds &lt;- shape.predictor(lmks$coords, x= PC, Intercept = FALSE, 
                         pred1 = min(PC), pred2 = max(PC))</code></pre>
<p>This returns an object that includes coordinates that have been
estimated for a specimen resembling the highest and lowest PCs in our
data set. We can visualize them using <a
href="">plotReftoTarget</a>:</p>
<div class="code">
<pre class="r custom-inline"><code>plotRefToTarget(lmks$consensus, preds$pred1)</code></pre>
</div>
<p><img src="4.3-shapepred_files/figure-html/PCvis1-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="code">
<pre class="r custom-inline"><code>plotRefToTarget(lmks$consensus, preds$pred2)</code></pre>
</div>
<p><img src="4.3-shapepred_files/figure-html/PCvis2-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="shape-prediction-using-allometric-relationship"
class="section level4">
<h4>2: <strong>Shape Prediction Using Allometric
Relationship</strong></h4>
<p>Allometric relationships can also be used as predictors for this
function, and can be implemented in several ways. The first is simply
using log centroid sizes of our dataset:</p>
<pre class="r custom-inline"><code>preds &lt;- shape.predictor(lmks$coords, x= log(lmks$Csize), 
                         Intercept = TRUE, 
                         predmin = min(log(lmks$Csize)), 
                         predmax = max(log(lmks$Csize))) </code></pre>
<div class="code">
<pre class="r custom-inline"><code>plotRefToTarget(lmks$consensus, preds$predmin, mag=3)
plotRefToTarget(lmks$consensus, preds$predmax, mag=3)</code></pre>
</div>
<p><img src="4.3-shapepred_files/figure-html/allovis-1.png" width="672" style="display: block; margin: auto;" /><img src="4.3-shapepred_files/figure-html/allovis-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Alternatively, implementing allometric relationships as predictors
can be done using the RegScore or PredLine methods from
<code>procD.lm</code>. We illustrate an example below, but see the
tutorial on <a href="">plotting allometry</a> for more details on these
methods.</p>
<p>First, we generate our fit:</p>
<pre class="r custom-inline"><code>gdf &lt;- geomorph.data.frame(lmks)
plethAllometry &lt;- procD.lm(coords ~ log(Csize), data=gdf)</code></pre>
<p>Then implement the function and visualize using
<code>plotRefToTarget</code></p>
<div class="code">
<pre class="r custom-inline"><code>allom.plot &lt;- plot(plethAllometry, type = &quot;regression&quot;, predictor = log(gdf$Csize),
                    reg.type =&quot;RegScore&quot;)
preds &lt;- shape.predictor(plethAllometry$GM$fitted, 
                        x= allom.plot$RegScore, Intercept = FALSE, 
                        predmin = min(allom.plot$RegScore), 
                        predmax = max(allom.plot$RegScore))

plotRefToTarget(lmks$consensus, preds$predmin, mag=3)
plotRefToTarget(lmks$consensus, preds$predmax, mag=3)</code></pre>
</div>
<p><img src="4.3-shapepred_files/figure-html/regscore-1.png" width="672" style="display: block; margin: auto;" /><img src="4.3-shapepred_files/figure-html/regscore-2.png" width="672" style="display: block; margin: auto;" /><img src="4.3-shapepred_files/figure-html/regscore-3.png" width="672" style="display: block; margin: auto;" /></p>
<div class="code">
<pre class="r custom-inline"><code>allom.plot &lt;- plot(plethAllometry, type = &quot;regression&quot;,predictor = log(gdf$Csize),
                    reg.type =&quot;PredLine&quot;)

preds &lt;- shape.predictor(plethAllometry$GM$fitted,
                         x= allom.plot$PredLine, Intercept = FALSE,
                         predmin = min(allom.plot$PredLine),
                         predmax = max(allom.plot$PredLine))

plotRefToTarget(lmks$consensus, preds$predmin, mag=3)
plotRefToTarget(lmks$consensus, preds$predmax, mag=3)</code></pre>
</div>
<p><img src="4.3-shapepred_files/figure-html/predline-1.png" width="672" style="display: block; margin: auto;" /><img src="4.3-shapepred_files/figure-html/predline-2.png" width="672" style="display: block; margin: auto;" /><img src="4.3-shapepred_files/figure-html/predline-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="shape-prediction-using-two-block-pls" class="section level4">
<h4>3: <strong>Shape Prediction Using Two-block PLS</strong></h4>
<p>Finally, results of a <a href="">two-block PLS</a> can be used as a
predictor for this function. Results of the PLS must be input into the
function as scalars (single values as opposed to vectors) that are taken
from a PLS plot. By way of example, we plot the results of a PLS
analysis between plethodontid head shape and food use data. Please see
the tutorial on <a href="">PLS</a> for details how to generate these
results:</p>
<div class="code">
<pre class="r custom-inline"><code>PLS &lt;-two.b.pls(A1 = plethShapeFood$food, A2 = plethfood$coords, iter=999, print.progress = F) 
plot(PLS)</code></pre>
</div>
<div class="code">
<pre class="r custom-inline"><code>plot(PLS)</code></pre>
</div>
<p><img src="4.3-shapepred_files/figure-html/PLSplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Values found along the axis of PLS 1 will be used as our scalars for
<code>shape.predictor</code>:</p>
<pre class="r custom-inline"><code>preds &lt;- shape.predictor(plethfood$coords, plethShapeFood$food, 
                         Intercept = FALSE,
                         method = &quot;PLS&quot;,
                         pred1 = 2, pred2 = -4, pred3 = 2.5)</code></pre>
<p>We can then visualize the results as normal:</p>
<div class="code">
<pre class="r custom-inline"><code>plotRefToTarget(plethfood$consensus, preds$pred1, mag=2)
plotRefToTarget(plethfood$consensus, preds$pred2, mag=2)
plotRefToTarget(plethfood$consensus, preds$pred3, mag=2)</code></pre>
</div>
<p><img src="4.3-shapepred_files/figure-html/PLSvis-1.png" width="672" style="display: block; margin: auto;" /><img src="4.3-shapepred_files/figure-html/PLSvis-2.png" width="672" style="display: block; margin: auto;" /><img src="4.3-shapepred_files/figure-html/PLSvis-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="shape-prediction-using-anova-and-design-matrix"
class="section level4">
<h4>4: <strong>Shape Prediction Using ANOVA and Design
Matrix</strong></h4>
<p>Results of a Procrustes ANOVA, run using <code>procD.lm</code> can,
in conjunction with a design matrix and <code>shape.predictor</code>, be
used to visualize shape differences among groups in a dataset. This
example assumes that the user has already generated their fit using
<code>procD.lm</code>. More details on performing a Procrustes ANOVA can
be found in the <a href="">ANOVA</a> tutorial, as well as the <a
href="">Group Comparisons</a> workflow.</p>
<pre class="r custom-inline"><code>X &lt;- fit$X
X &lt;- X[,-1]
symJord &lt;- c(0,1,0) # design for P. Jordani in sympatry
alloJord &lt;- c(0,0,0) # design for P. Jordani in allopatry
pred &lt;- shape.predictor(arrayspecs(fit$fitted, 12,2), x = X, symJord=symJord, alloJord=alloJord, 
                        Intercept = T)</code></pre>
<p>As arguments for the function we include the fitted values generated
from our Procrustes ANOVA (fix$X), as well as the model matrix of the
same analysis (X). This matrix indicates which individual belongs to
which subgroup of our data using 1s and 0s. The final two arguments here
are the specific groups (among our total potential combinations of
factors) for which we want to generate visual comparisons.</p>
<p>Then, we can visualize the difference between these two groups in a
similar fashion to the previous methods:</p>
<pre class="r custom-inline"><code>plotRefToTarget(lmks$consensus, pred$symJord, mag=2)</code></pre>
<p><img src="4.3-shapepred_files/figure-html/ANOVAvis-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r custom-inline"><code>plotRefToTarget(lmks$consensus, pred$alloJord, mag=2)</code></pre>
<p><img src="4.3-shapepred_files/figure-html/ANOVAvis-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
